src/problems/FoamProblem.C

```diff
@@ -372,3 +375,83 @@ FoamProblem::verifyFoamVariables()
   }
   vt.print(_console);
 }
+
+// Create comma separated list from vector
+template <typename StrType>
+inline std::string
+listFromVector(std::vector<StrType> vec, StrType sep = ", ")
+{
+  if (vec.size() == 0)
+    return std::string();
+  else if (vec.size() == 1)
+    return vec.at(0);
+
+  std::string str;
+  auto binary_op = [&](const std::string & acc, const std::string & it) { return acc + sep + it; };
+  std::accumulate(vec.begin(), vec.end(), str, binary_op);
+  return str;
+}
+
+void
+FoamProblem::verifyFoamBCs()
+{
+  // Get list of all variables used by all BCs
+  std::vector<std::string> variables(_foam_bcs.size());
+  for (auto & bc : _foam_bcs)
+    variables.push_back(bc->foamVariable());
+
+  std::set<std::string> unique_vars(variables.begin(), variables.end());
```

Comment:
Can you insert directly into the set, rather than creating a temporary vector?

```suggestion
  std::set<std::string> unique_vars;
  for (const auto & bc : _foam_bcs)
  {
    unique_vars.insert(bc->foamVariable());
  }
```